### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by malte-b**
*Description of changes:*
Add descriptions for all analyzers.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: analyzerDoc.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+## ApproxCountDistinct
+    Compute approximated count distinct with [HyperLogLogPlusPlus](https://en.wikipedia.org/wiki/HyperLogLog#HLL++)
+## ApproxQuantile
```
Comment (velramiir) [inline analyzerDoc.md]: According to wikipedia, this is not called the x-quantile, but the x-percentile. But his is also different in Amazon's code

File: analyzerDoc.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    Approximate quantile analyzer. 
+    Yields the x-quantile; a value, for which it holds true, that exaclty x percent of the total values are smaller than the yielded value.
+    The quantile x can be passed to the analyzer as parameter, x has to be a number between 0 and 1. Choosing 0.5 will yield the median.
+    The allowed relative error compared to the exact quantile can be configured with `relativeError` parameter. A `relativeError` = 0.0 would yield the exact quantile while increasing the computational load.
+## ApproxQuantiles
+    The same as ApproxQuantile, can be given a sequence of quantiles.
+## Completeness
+    Completeness is the fraction of the number of non-null values devided by the number of all values in a column.
```
Comment (velramiir) [inline analyzerDoc.md]:
```suggestion
    Completeness is the fraction of the number of non-null values divided by the number of all values in a column.
```

File: analyzerDoc.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+## Compliance
+    Compliance is a measure of the fraction of rows that complies with the given column constraint.
```
Comment (velramiir) [inline analyzerDoc.md]:
```suggestion
    Compliance is a measure of the fraction of rows that complies with the given constraint. This constraint may span multiple columns.
```

File: analyzerDoc.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    E.g if the constraint is "att1>3" and data frame has 5 rows with att1 column value greater than
```
Comment (velramiir) [inline analyzerDoc.md]:
```suggestion
    e.g. Given the constraint is "att1 > 3" and the data frame has 5 rows with an att1 column value greater than
```

File: analyzerDoc.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    3 and 10 rows under 3; a DoubleMetric would be returned with 0.33 value
```
Comment (velramiir) [inline analyzerDoc.md]:
```suggestion
    3 and 10 rows with a value lower than 3, a DoubleMetric of 0.33 would be returned.
```

File: analyzerDoc.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+## Correlation
+    Computes the [pearson correlation coefficient](https://en.wikipedia.org/wiki/Pearson_correlation_coefficient) between the two given columns.
+    It has a value between -1 and +1, where +1 is total positive linear correlation, 0 is no linear correlation, and −1 is total negative linear correlation.
```
Comment (velramiir) [inline analyzerDoc.md]:
```suggestion
    It returns a value between -1 and +1, where +1 is total positive linear correlation, 0 is no linear correlation, and −1 is total negative linear correlation.
```

File: analyzerDoc.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+## CountDistinct
+    Number of distinct values in the column.
+## DataType
+    Yields a distribution map, including the overall number of values of each datatype and the percentage of each datatype. 
+    Possible datatypes are Boolean, Fractional, Integral, Unknown and String.
```
Comment (velramiir) [inline analyzerDoc.md]:
```suggestion
    Possible datatypes are Boolean, Fractional, Integral, and String. If the type can't be determined, Unknown is returned.
```

File: analyzerDoc.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+## Distinctness
+    Distinctness is the fraction of the number of distinct values devided by the number of all values of a column. Can be given a sequence of columns.
+## Entropy
+    [Entropy](https://en.wikipedia.org/wiki/Entropy_(information_theory)) is a measure of the level of information contained in a message. Given the probability distribution over values in a column, it describes how many bits are required to identify a value.
```
Comment (velramiir) [inline analyzerDoc.md]:
```suggestion
    [Entropy](https://en.wikipedia.org/wiki/Entropy_(information_theory)) is a measure of the level of information contained in a message. Given the probability distribution over values in a column, it describes how many bits are required to identify a value. A highly diverse column will result in high entropy. Few different values will result in low entropy.
```

File: analyzerDoc.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+## Histogram
+    Divides the columns of the values in the given column into buckets. The number of buckets has to be given by the user and has to be smaller than or euqal to 1000. Furthermore the user has to give a user defined function as an input parameter that is applied to the bins.
```
Comment (velramiir) [inline analyzerDoc.md]: I don't get what the user defined function is supposed to be.

Comment (velramiir) [inline analyzerDoc.md]:
```suggestion
    Computes the histogram for the given columns. Small buckets may be filtered in the output. Furthermore the user has to specify a user defined function as an input parameter that is applied to the bins.
```

File: analyzerDoc.md [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+## Maximum
+    Returns the largest value in the given column. Works only for numerical columns. Only for numeric columns.
```
Comment (velramiir) [inline analyzerDoc.md]:
```suggestion
    Returns the largest value in the given column. Only for numeric columns.
```

File: analyzerDoc.md [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+## Mean
+    This is the average over all values of the given column. Therefore it is the fraction of the Sum and the Size of the column. Only for numeric columns.
```
Comment (velramiir) [inline analyzerDoc.md]:
```suggestion
    This is the average over all values of the given column. Therefore it is the fraction of the sum and the size of the column. Only for numeric columns.
```

File: analyzerDoc.md [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+## Median
+    Returns the value in the middle of the given column. This is more robust against outliers than the mean. It is equal to the 2-quantile.
```
Comment (velramiir) [inline analyzerDoc.md]:
```suggestion
    For n rows returns the value with index n/2 in the specified column if sorted. If n is odd the mean between the surrounding indices is computed. This is more robust against outliers than the mean. It is equal to the 2-quantile.
```

Comment (velramiir) [reply analyzerDoc.md]: Would also be 2-percentile according to wikipedia

File: analyzerDoc.md [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+## Minimum
+    Returns the smallest value in the given column. Only for numeric columns.
+## Mode
+    This is the most common value in the given column.
```
Comment (velramiir) [inline analyzerDoc.md]:
```suggestion
    This is the most common value in the given column. Not necessarily unique.
```

File: analyzerDoc.md [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+## MutualInformation
+    It measures how much having information about one column reveals about aonther. If both columns are independent, the mutual information is zero. If one of the columns determines all values of the other column and the other way around, there is a functional dependency between the columns. In that case, the mutual information is equal to either of both columns' Entropy.
```
Comment (velramiir) [inline analyzerDoc.md]:
```suggestion
    Measure of how much information one column reveals about another. If both columns are independent, the mutual information is zero. If one of the columns determines all values of the other column and the other way around, there is a functional dependency between the columns. In that case, the mutual information is equal to either of both columns' entropy.
```

**Review by velramiir — CHANGES_REQUESTED** [review]
Would be nice to establish a common format for all analyzers. e.g.

- intuitive description
- input parameters
- return value (value range)
- formula
- allowed column types

File: analyzerDoc.md (hunk 1 above)
Comment (malte-b) [reply analyzerDoc.md]: That's true but we should keep it consistent with Amazon's notation for now

**Comment by malte-b** [comment]
@velramiir please review the changes again. I tried to implement all your suggestions.  A common format for all analyzers is generally a good idea, but this should suffice for now imo. Additionally that would create a big overhead that could lead to less clarity for certain analyzers.

File: doc/analyzerDoc.md [hunk 15]
```diff
@@ -0,0 +1,54 @@
+## ApproxCountDistinct
+Compute approximated count distinct with [HyperLogLogPlusPlus](https://en.wikipedia.org/wiki/HyperLogLog#HLL++)
+## ApproxQuantile
+Approximate quantile analyzer. 
+Yields the x-quantile; a value, for which it holds true, that exaclty x percent of the total values are smaller than the yielded value.
```
Comment (velramiir) [inline doc/analyzerDoc.md]:
```suggestion
Yields the x-quantile; a value, for which it holds true, that exactly x percent of the total values are smaller than the yielded value.
```

File: doc/analyzerDoc.md [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+Number of distinct values in the column.
+## DataType
+Yields a distribution map, including the overall number of values of each datatype and the percentage of each datatype. 
+Possible datatypes are Boolean, Fractional, Integral and String. If the type can't be determined, Unknown is returned.
+## Distinctness
+Distinctness is the fraction of the number of distinct values devided by the number of all values of a column. Can be given a sequence of columns.
+## Entropy
+[Entropy](https://en.wikipedia.org/wiki/Entropy_(information_theory)) is a measure of the level of information contained in a message. Given the probability distribution over values in a column, it describes how many bits are required to identify a value. A highly diverse column will result in high entropy. Few different values will result in low entropy.
+## Histogram
+Divides the columns of the values in the given column into buckets. The number of buckets has to be given by the user and has to be smaller than or euqal to 1000.
```
Comment (velramiir) [inline doc/analyzerDoc.md]: This is still not a precise description in my opinion. Choosing the number of buckets will be understood as specifying into how many buckets a continuous value range should be divided. What actually happens here is that values with low cardinality are filtered out and simply not included in the result. This is a totally different behaviour and it should be explicitly stated in the description.
Aditionally I find the first sentence confusing. The analyzer doesn't create buckets as far as I know. It simply counts the frequency of all occuring values within a column, even if they are numeric.

**Review by velramiir — CHANGES_REQUESTED** [review]

File: doc/analyzerDoc.md (hunk 16 above)
Comment (malte-b) [reply doc/analyzerDoc.md]: Fixed it

**Review by velramiir — APPROVED** [review]

**Pull request merged**